games-arcade/rocksndiamonds/patches/rocksndiamonds-3.3.1.2.patchset

```diff
@@ -30,10 +30,10 @@ index a67cd14..8493bb9 100644
  
  SRCS =	main.c		\
 -- 
-1.8.3.4
+2.7.0
 
 
-From 142ac2bc985591b85860f6fea82b5161fc4446e9 Mon Sep 17 00:00:00 2001
+From a90cc639c7268bb4ed9118aa691014ad61963161 Mon Sep 17 00:00:00 2001
```

Comment:
These SHA1 checksums are randomly generated by "haikuporter -e", and they are only used inside the work-xxx subdirectory.
So when haikuporter applies a patch, it uses the SHA1 checksums found in the patch it is using.

So:

* if the work directory was clean before applying a patchset, it won't make any difference if we use the newly gerenated SHA1 checksum or the one previously gerenated

* if the work directory was not clean but no patches had been applied yet, then the conclusion is the same as above.

* if the work directory was not clean and the patchset with the previous checksum had already been applied, then it won't have to read the patchset again, since the "patch applied" flag is already set. So the contents of the patchset can be anything.

I've been keeping the original SHA1 cjecksums for all patches I've been regenerating (when I need to update a recipe which has patchsets), and I can certify that it won't ever break anything. And even when some patches need small changes, I still keep the original checksums. This has never broken anything.
There are only benefits in keeping original SHA1 checksum: when you look for diffs (with "git diff" or "git log -p") then having logs that do not add unneeded noise is really nice.